src/main/java/net/yogstation/yogbot/ByondConnector.kt

```diff
@@ -27,12 +27,12 @@ class ByondConnector(private val config: ByondConfig) {
 				val outputStream = socket.getOutputStream()
 				outputStream.write(buffer.array())
 				val inputStream = socket.getInputStream()
-				val headerBuffer = ByteArray(4)
-				if (inputStream.read(headerBuffer) < 4) return YogResult.error("Returned insufficient data")
+				val headerBuffer = inputStream.readNBytes(4)
+				if (headerBuffer.size < 4) return YogResult.error("Returned insufficient data")
```

Comment:
Can this actually happen with readNBytes?

Reply:
End of stream is reached